src/components/map/MapService.js

```diff
@@ -405,19 +405,114 @@
       }
     };
 
-    this.$get = function($http, gaDefinePropertiesForLayer, gaMapClick,
-        gaMapUtils, gaGlobalOptions, $rootScope, $translate, gaStyleFactory,
-        gaStorage, gaNetworkStatus) {
+    this.$get = function($http, $q, $rootScope, $timeout, $translate,
+        gaDefinePropertiesForLayer, gaGlobalOptions, gaMapClick, gaMapUtils,
+        gaNetworkStatus, gaStorage, gaStyleFactory, gaUrlUtils) {
+
       // Create the parser
       var kmlFormat = new ol.format.KML({
         extractStyles: true,
         extractAttributes: true,
         defaultStyle: [gaStyleFactory.getStyle('kml')]
       });
-      var Kml = function(proxyUrl) {
-        /**
-         * Create a KML layer from a KML string
-         */
+
+      // Read a kml string then return a list of features.
+      var readFeatures = function(kml) {
+        // Replace all hrefs to prevent errors if image doesn't have
+        // CORS headers. Exception for google markers icons (lightblue.png,
+        // ltblue-dot.png, ltblu-pushpin.png, ...) to keep the OL3 magic for
+        // anchor origin.
+        // Test regex here: http://regex101.com/r/tF3vM0
+        // List of google icons: http://www.lass.it/Web/viewer.aspx?id=4
+        kml = kml.replace(
+          /<href>http(s?)(?!(s?):\/\/maps\.(?:google|gstatic)\.com.*(blue|green|orange|pink|purple|red|yellow|pushpin).*\.png)/g,
+          '<href>' + gaGlobalOptions.ogcproxyUrl + 'http'
+        );
+
+        var all = [];
+        var features = kmlFormat.readFeatures(kml);
+        var networkLinks = kmlFormat.readNetworkLinks(kml);
+        if (networkLinks.length) {
+          angular.forEach(networkLinks, function(networkLink) {
+            if (gaUrlUtils.isValid(networkLink.href)) {
+              all.push($http.get(networkLink.href).success(function(data) {
```

Comment:
Nothing. It's intentional. The user is responsible for setting good URL in his KML.